nodejs/aws-infra/experimental/ec2/securityGroup.ts

```diff
@@ -79,6 +79,32 @@ export class SecurityGroup extends pulumi.ComponentResource {
     }
 }
 
+export type SecurityGroupOrId = SecurityGroup | pulumi.Input<string>;
```

Comment:
this seems like a generally reasonable pattern for resources we want to allow passing either as rich objects, or as IDs to be rehydrated.  VPCs, subnets, etc. would likely all benefit from a similar approach.

Reply:
The name `XxxOrId` also seems better than ```XxxLike```